src/app/ReadClient.h

```diff
@@ -165,6 +165,17 @@ class ReadClient : public Messaging::ExchangeDelegate
          * SendAutoResubscribeRequest is not called, this function will not be called.
          */
         virtual void OnDeallocatePaths(ReadPrepareParams && aReadPrepareParams) {}
+
+        /**
+         * This function is invoked when using read/subscribeRequest, where the ReadClient would use cached cluster data version to
+         * update data version filter list
```

Comment:
This is completely inscrutable.  What does "update data version filter list" mean?  How does this interact with the data varsions provided via the read params?  What does the return value mean?

The documentation needs to be such that someone can read it and understand whether they should implement the function and if so how to implement it.

That said, why are we hanging this off the callback anyway?  Wouldn't it make more sense to have a way for an attribute cache to hand out the right thing to set on the read params and then whoever is using the cache as a read handler can do that?

Reply:
Updated, thanks